### ISSUE TITLE
[Bug]: Cannot edit Amount for Approved Tokens

### ISSUE THREAD

**Issue by seaona** [MetaMask/metamask-extension#16357]
### Describe the bug

**Problem**: I cannot edit the amount for the Approve Token screen. No error is displayed.

https://user-images.githubusercontent.com/54408225/199556355-3a17e2f4-a009-4407-950a-4327bb7412d8.mp4

### Steps to reproduce

BEFORE TESTING - in your .metamaskrc file, set `TOKEN_ALLOWANCE_IMPROVEMENTS=1` and then start the app
1. Unlock MM
2. Go to test dapp https://metamask.github.io/test-dapp/
3. Connect to test Dapp
4. Click Create Token and Accept MM prompt
5. Click Add Token and Accept MM prompt
6. Click Approve Token
7. Click Edit on MM prompt -- nothing happens

### Version

10.22.0

### Browser

Chrome

### Operating system

Linux

**Comment by bschorchit** [comment]
This feature is not fully completed yet. This PR https://github.com/MetaMask/metamask-extension/issues/15211 has yet to be merged for it. Once it's merged, when you click on `Edit` it will take you back to the new screen.

**Comment by seaona** [comment]
I remove the RC label, as per @bschorchit and @filipsekulic information. This issue can be closed once the Token Allowance PR is QAd and merged to develop :ok_hand:

**Comment by bschorchit** [comment]
https://github.com/MetaMask/metamask-extension/pull/16157 has now been merged, @seaona 😊

**Comment by seaona** [comment]
Ty @bschorchit if possible, I want to QA it before closing this issue

**Comment by seaona** [comment]
@bschorchit I see this issue fixed. However, I have a question related to the new behaviour.

Previously, when we call the Approve function from the Dapp, we used to grab the values for sender and amount, as per:
`approve(spender, amount)`
I see that now it's not the case anymore, and we always fill the amount from MM (being the default value 0). Is this the expected behaviour? Thanks!!

See example below with test-dapp, where we set an approve for 7 tokens, and on MM we see 0 as default.

https://user-images.githubusercontent.com/54408225/201345091-cd790f88-60b3-411c-8299-4667e7b655d5.mp4
